src/qtcwin.c

```diff
@@ -0,0 +1,1816 @@
+						qtcreclist.qtclines[currqtc].confirmed = 1;
+						qtcreclist.confirmed++;
+						if (trxmode == CWMODE) {
+						    sendmessage(qtc_recv_msgs[2]);
+						}
+						if (trxmode == SSBMODE) {
+						    play_file(qtc_phrecv_message[2]);
+						}
+					    }
+					    tfi = (activefield-3)%3;
+					    //activefield++;	// go to next line time field
+					    activefield += (3-tfi);
+					    showfield(activefield-(3-tfi));
+					}
+					showfield(activefield);
+				    }
+				}
+				else if (qtcreclist.qtclines[currqtc].status == 1 && qtcreclist.qtclines[currqtc].confirmed != 1) {
+				    if (trxmode == CWMODE) {
+					sendmessage(qtc_recv_msgs[7]);
+				    }
+				    if (trxmode == SSBMODE) {
+					play_file(qtc_phrecv_message[7]);
+				    }
+				    if (trxmode == DIGIMODE) {
+					char *str = g_strdup_printf("%s %02d %02d\n", g_strchomp(qtc_recv_msgs[7]), currqtc+1, currqtc+1);
+					sendmessage(str);
+					g_free(str);
+				    }
+				}
+
+				if (*qtccount > 0 && qtcreclist.confirmed == *qtccount) {
+				    if (qtcreclist.sentcfmall == 0) {
+					qtcreclist.sentcfmall = 1;
+					if (log_recv_qtc_to_disk(nr_qsos) == 0) {
+					    // TODO
+					    // send 'CFM all' to station
+					  if (qtcrec_record == 1 && record_run > -1) {
+					      strcpy(reccommand, "pkill -SIGINT -n ");
+					      strcat(reccommand, qtcrec_record_command_shutdown);
+					      system(reccommand);
+					      record_run = -1;
+					  }
+					  if (trxmode == DIGIMODE || trxmode == CWMODE) {
+						sendmessage(qtc_recv_msgs[9]);
+					    }
+					    if (trxmode == SSBMODE) {
+						play_file(qtc_phrecv_message[9]);
+					    }
+					    // TODO
+					}
+				    }
+				    x = 27;	// close the window
+				}
+			    }
+			}
+			if (direction == SEND && trxmode != DIGIMODE) {
+			    if (qtclist.qtclines[activefield-3].sent == 0) {
+				qtclist.qtclines[activefield-3].sent = 1;
+				get_time();
+				tempc[0] = '\0';
+				strftime(tempc, 40, "%d-%b-%y %H:%M", time_ptr);
+				strncpy(qtclist.qtclines[activefield-3].senttime, tempc, 15);
+				qtclist.qtclines[activefield-3].senttime[15] = '\0';
+				qtclist.totalsent++;
+			    }
+			    tempc[0] = '\0';
+			    strip_spaces(qtclist.qtclines[activefield-3].qtc, tempc);
+			    data_ready = 1;
+			    if (trxmode == CWMODE) {
+				sendmessage(tempc);
+			    }
+
+			    mvwprintw(qtcwin, activefield, 30, "*");
+			    qtclist.qtclines[activefield-3].flag = 1;
+			    // scroll down if not at end of qtclist:
+			    if (activefield-3 < *qtccount-1) {
+				wattrset(qtcwin, line_normal);
+				mvwprintw(qtcwin, activefield, 4, "%s", qtclist.qtclines[(activefield-3)].qtc);
+				activefield++;
+				wattrset(qtcwin, line_currinverted);
+				mvwprintw(qtcwin, activefield, 4, "%s", qtclist.qtclines[(activefield-3)].qtc);
+			    }
+			    if (*qtccount > 0 && qtclist.totalsent == *qtccount) {
+				wattrset(qtcwin, line_inverted);
+				mvwprintw(qtcwin, 2, 11, "CTRL+S to SAVE!");
+				refreshp();
+				show_help_msg(6);
```

Comment:
Oh, sorry - the length of help_rec_msgs[] is 6, that's correct. But the help_send_msgs[] is 7, and there is a condition before that snippet:
```C
510.    if (direction == SEND && trxmode != DIGIMODE) {
...
537.        if (*qtccount > 0 && qtclist.totalsent == *qtccount) {
...
541.            show_help_msg(6)
```
So, I think there couldn't be overflowed. Anyway, this part of code executed when SENDER station sent all line of QTC in DIGIMODE - shows the message: "Press CTRL+S to save"